src/Symfony/Component/Validator/Mapping/GenericMetadata.php

```diff
@@ -213,6 +236,14 @@ public function getTraversalStrategy()
         return $this->traversalStrategy;
     }
 
+    /**
+     * @see AutoMappingStrategy
+     */
+    public function getAutoMappingStrategy(): int
```

Comment:
Worth adding to the interface (`@method` to get the deprec), or adding a new `AutoMappingMetadataInterface`?
AFAIK, implementations all relies on `GenericMetadata` classes, so might only be an implementation detail.